db/memtable.cc

```diff
@@ -416,24 +419,38 @@ InternalIterator* MemTable::NewIterator(const ReadOptions& read_options,
   return new (mem) MemTableIterator(*this, read_options, arena);
 }
 
+void MemTable::RebuildFragmentedTombstones(const ReadOptions& read_options) {
+  auto* unfragmented_iter = new MemTableIterator(
+      *this, read_options, nullptr /* arena */, true /* use_range_del_table */);
+
+  if (fragmented_tombstones_invalidated) {
```

Comment:
It doesn't look safe to access `fragmented_tombstones_invalidated` without an exclusive lock. Am I misunderstanding something about the way this is supposed to work? It seems like this is called with `fragmented_tombstones_mut.ReadLock` held.

Reply:
Yeah.  You are right.